### ISSUE TITLE
Query regarding faceting

### ISSUE THREAD

**Issue by drsantam** [davidsjoberg/ggsankey#11]
It would be nice if the plots could be faceted using facet grid or other ggplot faceting methods. Is there a way to currently facet plots ? I am currently using cowplot and patchwork to assemble multiple plots into groups.

**Comment by davidsjoberg** [comment]
It doesnt work? Do you have an exemple?

Bästa hälsningar 
David

**Comment by drsantam** [comment]
Yes I am attaching an example dataset. In this dataset, each row corresponds to observations made on a single unit (say patient) and the arm represents the trial arm. When we use the make_long command, on this dataframe we have to select the columns called month_1, month_2 and month_3. The dataframe created by this command does not retain the arm and therefore I don't see a way to facet the plots by the arm. Please let me know if this is actually possible. 
[example_data.csv](https://github.com/davidsjoberg/ggsankey/files/7212451/example_data.csv)

**Comment by jpaulitz** [comment]
I was curious how this might work. 
You can use `group_modify()` from the purr package on a grouped dataframe to keep the grouping variable when using `make_long()`. Then faceting works as expected. 
Using your example data as df this code should do what you want?
```
df %>%
  group_by(arm) %>%
  group_modify(~make_long(.x,month_1:month_3)) %>%
  ggplot(aes(x = x, next_x = next_x, node = node, next_node = next_node, fill = factor(node))) +
  geom_sankey() +
  facet_grid(~arm)
```
![image](https://user-images.githubusercontent.com/10252911/134646982-86803d5f-a59d-456a-9449-9c7d5db59a59.png)

**Comment by drsantam** [comment]
Thanks I can confirm it works at my end too.

**Comment by stragu** [comment]
As I see it, this could be closed as it was about data preparation before using ggsankey. Is that correct?

**Comment by drsantam** [comment]
Yes. Thank you.
